### PR TITLE
Update python side from pipenv -> poetry

### PR DESCRIPTION
**Pull request by mathcolo**
In preparing to do the new train run count stuff on the train tracker I figured I may as well do this

### CODE REVIEW

**Review by idreyn — APPROVED** [review]
I'm supportive of moving off `pipenv` and this looks pretty comprehensive! though I defer to @nathan-weinberg on anything related to CI/CD.

File: devops/deploy-on-lightsail.yml
Comment (nathan-weinberg) [inline devops/deploy-on-lightsail.yml:43]: @mathcolo why the change from from `update` to `install`?

File: .github/workflows/lint.yml
Comment (nathan-weinberg) [inline .github/workflows/lint.yml:21]: Can this not be `pip` installed? (Fine if it can't!)

File: pyproject.toml
Comment (nathan-weinberg) [inline pyproject.toml:3]:
```suggestion
version = "1.0.0"
```
Should this match the git tag version?

File: devops/deploy-on-lightsail.yml
Comment (mathcolo) [reply devops/deploy-on-lightsail.yml:43]: `npm update` actually changes package.json, which is useful as an action I might take manually, but is not something deploy should be doing

For that matter `npm ci` is actually probably the most appropriate command here! (https://docs.npmjs.com/cli/v6/commands/npm-ci)

File: .github/workflows/lint.yml
Comment (mathcolo) [reply .github/workflows/lint.yml:21]: It can be, but the official documentation strongly recommends not doing so https://python-poetry.org/docs/master/#installing-with-pip

File: devops/deploy-on-lightsail.yml
Comment (nathan-weinberg) [reply devops/deploy-on-lightsail.yml:43]: @mathcolo sounds good, i'd be open to trying it out! perhaps it could be used in our other CD definitions as well if it works here

File: pyproject.toml
Comment (mathcolo) [reply pyproject.toml:3]: It could, but doesn't really matter 👍 The version number here has nearly no function

Comment (nathan-weinberg) [reply pyproject.toml:3]: no worries, the `package.json` matches which is probably more important - certainly not going to block merging on this

**Review by nathan-weinberg — APPROVED** [review]
Let 'er rip!

**Comment by bpblanken** [comment]
I approve of this though nobody asked me!

**Comment by mathcolo** [comment]
LOL hey ben!

**Pull request merged**
